src/index.ts

```diff
@@ -263,36 +263,38 @@ interface RuntimeInfo {
  * @return Promise of infomation of runtime of `target`.
  */
 async function searchScriptRuntime (target: string, opts: InternalOptions): Promise<RuntimeInfo> {
+  let shebang;
   try {
     const data = await opts.fs_.readFile(target, 'utf8')
 
     // First, check if the bin is a #! of some sort.
     const firstLine = data.trim().split(/\r*\n/)[0]
-    const shebang = firstLine.match(shebangExpr)
-    if (!shebang) {
-      // If not, infer script type from its extension.
-      // If the inference fails, it's something that'll be compiled, or some other
-      // sort of script, and just call it directly.
-      const targetExtension = path.extname(target).toLowerCase()
-      return {
-        // undefined if extension is unknown but it's converted to null.
-        program: extensionToProgramMap.get(targetExtension) || null,
-        additionalArgs: ''
+    shebang = firstLine.match(shebangExpr)
+  } catch (err: any) {
+    if (isWindows() && err.code === 'ENOENT') {
+      if (await opts.fs_.stat(`${target}${getExeExtension()}`)) {
+        return {
+          program: null,
+          additionalArgs: '',
+        }
       }
     }
+    if (err.code !== 'ENOENT') throw err
```

Comment:
The operation to compare `err.code` to `ENOENT` could be executed twice on Windows. You can optimize it by moving `if (err.code !== 'ENOENT') throw err` up to right under `catch`.